src/server/pps/server/kube_driver.go

```diff
@@ -173,3 +173,15 @@ func (kd *kubeDriver) WatchPipelinePods(ctx context.Context) (<-chan watch.Event
 	}
 	return kubePipelineWatch.ResultChan(), kubePipelineWatch.Stop, nil
 }
+
+func (kd *kubeDriver) GetWorkerImageID(ctx context.Context, podName string) (string, error) {
+	pod, err := kd.kubeClient.CoreV1().Pods(kd.namespace).Get(
+		ctx,
+		podName,
+		metav1.GetOptions{})
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get kubernetes pod")
+	}
+
+	return pod.Status.ContainerStatuses[1].ImageID, nil // second container should be the pipeline worker, first should be pachd
```

Comment:
We should probably iterate through the containers and check the names to be sure we got the right one. We could probably also make this function more general by parameterizing the container name.